File: src/main/java/com/example/employee_system/Service/EmployeeService.java
package com.example.employee_system.Service;

import com.example.employee_system.Model.Employee;

import java.util.List;

public interface EmployeeService {

    Employee createEmployee(Employee employee);

    List<Employee> getAllEmployees();

    boolean deleteEmployee(Long id);

    Employee getEmployeeById(Long id);

    Employee updateEmployee(Long id, Employee employee);
}

File: src/main/frontend/src/services/EmployeeService.js
import axios from "axios";

const EMPLOYEE_API_BASE_URL = "http://localhost:8080/api/v1/employees";

class EmployeeService {

    saveEmployee(Employee) {
        return axios.post(EMPLOYEE_API_BASE_URL, Employee);
    }
    getAllEmployees() {
        return axios.get(EMPLOYEE_API_BASE_URL);
    }
    deleteEmployee(id) {
        return axios.delete(EMPLOYEE_API_BASE_URL + "/" + id);
    }

    getEmployeeById(id) {
        return axios.get(EMPLOYEE_API_BASE_URL + "/" + id);
    }

    updateEmployee(Employee, id) {
        return axios.get(EMPLOYEE_API_BASE_URL + "/" + id, Employee);
    }

}

export default new EmployeeService();